lib/expression.ex

```diff
@@ -83,9 +83,11 @@ defmodule Expression do
     String.replace(expression, ~r/@([a-z]+)(\(|\.)?/i, "@@\\g{1}\\g{2}")
   end
 
-  @spec parse!(String.t() | Number.t()) :: Keyword.t()
+  @spec parse!(String.t() | Number.t() | list()) :: Keyword.t()
   def parse!(expression) when is_number(expression), do: to_string(expression) |> parse!()
 
+  def parse!(expression_ast) when is_list(expression_ast), do: expression_ast
```

Comment:
Is there a better way to match Expressions ASTs?

Reply:
right now there isn't :/

Reply:
Perhaps you can write a `defguard` function to make sure the given payload contains an `:expression` attribute and the expression payload is valid.

Reply:
Otherwise, it can be possible to just give a random list such as `["hola", "mundo"]` which is not valid.

Reply:
Good idea @santiagocardo! Will do tomorrow!